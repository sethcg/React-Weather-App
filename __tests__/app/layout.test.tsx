import { render, waitFor } from '@testing-library/react'
import RootLayout from '../../src/app/layout.tsx'

// THIS IGNORES THE "validateDOMNesting" CONSOLE.ERROR,
// THIS IS AN EXPECTED ERROR, BECAUSE RENDER() PUTS THE <HTML> ROOT IN A <DIV> TAG
jest.spyOn(global.console, 'error').mockImplementation(() => jest.fn())

describe('Root Snapshot', () => {
  it('Renders root layout unchanged', async () => {
    const mockChildren = <div>Children component</div>
    const { container } = await waitFor(() => render(<RootLayout>{mockChildren}</RootLayout>))

    expect(container).toMatchSnapshot()
  })
})
