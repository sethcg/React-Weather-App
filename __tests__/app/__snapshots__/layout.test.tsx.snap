// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Snapshot Renders root layout unchanged 1`] = `
<div>
  <div>
    Children component
  </div>
</div>
`;
